lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -1453,12 +1459,14 @@ instance FromHttpApiData ApiPoolId where
         | t == "*" =
             Right ApiPoolIdPlaceholder
         | otherwise =
-            bimap pretty ApiPoolId (fromText t)
+            case fmap ApiPoolId (fromText t) of
+              Left _  -> bimap (T.pack . displayException) ApiPoolId . decodePoolIdBech32 $ t
+              Right r -> Right r
 
 instance ToHttpApiData ApiPoolId where
     toUrlPiece = \case
         ApiPoolIdPlaceholder -> "*"
-        ApiPoolId pid -> toText pid
+        ApiPoolId pid -> encodePoolIdBech32 pid
```

Comment:
Make sure to also fix the associated `FromHttpApiData` as well!

Reply:
That is part of this PR already https://github.com/input-output-hk/cardano-wallet/pull/2055/files#diff-e1113f15531f8432f00f243428125671R1457-R1464